src/components/repl.tsx

```diff
@@ -271,7 +325,16 @@ const Repl: ReplProps = (props) => {
         <Show when={props.current}>
           <Editor
             url={`file:///${props.id}/${props.current}`}
-            onDocChange={() => compile()}
+            onDocChange={(code: string) => {
+              if (props.current?.startsWith('data_')) {
+                if (props.current == 'data_import_map.json') {
+                  const newImportMap = JSON.parse(code);
```

Comment:
Seems like it might be even cleaner to store the importmap as a string in the signal, that way you only have to parse when diffing it?

Reply:
This means that the output import map would have formatting applied to it as that's what the editor would return, and this breaks import maps as the quotes are escaped